Clarify migration script dependencies and make aware of the mysql_connector_python error

marwyg opened the pull request:
I updated the docs, so people hopefully don't run into the weird "UTF-8" error, when executing the 11-to-12 database migration script.

mliradelc reviewed (approved):
Good for me

marwyg committed:
Update migration script doc to make aware of the mysql_connector_python error

lkiesow commented on docs/guides/admin/docs/upgrade.md:
```diff
@@ -49,10 +49,19 @@ You can do this in a virtual Python environment if you do not want to install th
 pip install psycopg2-binary
-# or for MariaDB
+
+# or for MariaDB:
 pip install mysql-connector-python
+# NOTE: please don't install version 8.0.30, because the migration
+# script runs into errors with this version. Please install
+# version 8.0.29 or try a newer version than 8.0.30 if it exists. 
+# You can install a specific version with: 
+# 'pip install mysql_connector_python==8.0.29'
```
Why not just recommend installing that version?

```suggestion
# or for MariaDB:
pip install mysql-connector-python==8.0.29'
```

lkiesow reviewed (commented):
The documentation in the script still tells you to install the latest version:

https://github.com/opencast/opencast/blob/bb76ce70bd7e695a4a132c279e3ed629818c52d1/docs/upgrade/11_to_12/workflows_mariadb.py#L3-L4

marwyg commented:
I made two PRs: https://github.com/opencast/opencast/pull/4033 and to be honest, I have no idea why I thought, this is a good idea. So I added both doc changes into this PR. I also changed the installation command, so you are always installing version 8.0.29 of the module.

marwyg commented:
https://github.com/opencast/opencast/pull/4033#issuecomment-1198995360
@mliradelc I tried to put a try-catch around the connection, but had trouble with it. Could you add it, like you suggested it in the other PR?

mliradelc committed:
Adds try-catch

mliradelc commented:
@marwyg That should make it 😄

marwyg commented:
Thank you :grin: :+1:
